src/sources/util/http/prelude.rs

```diff
@@ -218,6 +227,10 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
     }
 }
 
+#[derive(Clone)]
+#[repr(transparent)]
```

Comment:
I'm not super familiar with this macro, can you expand on why it's needed?

Reply:
It's just a minor memory optimization. Basically, this suggests the compiler that the type itself doesn't have any size. So, while performing compiler optimization this types gets abstracted as the inner type.

Reply:
Here is a reference if you wish to read about it: https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent